tests/v2/validation/fleet/provisioning/networkisolation.go

```diff
@@ -0,0 +1,231 @@
+package provisioning
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/rancher/tests/v2/actions/kubeapi/namespaces"
+	"github.com/rancher/rancher/tests/v2/actions/kubeapi/workloads/deployments"
+	"github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/workloads/pods"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/kubeconfig"
+	"github.com/rancher/shepherd/extensions/unstructured"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	extensionpods "github.com/rancher/shepherd/extensions/workloads/pods"
+)
+
+const defaultNamespace = "default"
+
+// testPNI is a set of in-order instructions that gets the IP from a pod in an existing deployment/project/namespace,
+// then creates a new deployment in the default project/namespace which tries to ping the existing pod.
+// This errors if the new pod can ping the existing one, meaning that PNI is not working.
+// Not designed to work with a deployment already in the default namespace!
+func testPNI(client *rancher.Client, clusterName, deploymentNamespace, deploymentName string) error {
+	// get one of the fleet deployment's pod IPs to use in the ping command later
+	fleetPodNames, err := pods.GetPodNamesFromDeployment(client, clusterName, deploymentNamespace, deploymentName)
+	if err != nil {
+		return err
+	}
+
+	firstPodObject, err := pods.GetPodByName(client, clusterName, deploymentNamespace, fleetPodNames[0])
+	if err != nil {
+		return err
+	}
+
+	corePod := &corev1.Pod{}
+	err = steveV1.ConvertToK8sType(firstPodObject, corePod)
+	if err != nil {
+		return err
+	}
+
+	// create a new workload / pod to ping that's hardened-compliant
+	falseP := false
+	trueP := true
+	userID := int64(1000)
+	container := corev1.Container{
+		Name:            "test-pni" + namegenerator.RandStringLower(3),
+		Image:           "redis:7-alpine",
+		ImagePullPolicy: corev1.PullAlways,
+		VolumeMounts:    nil,
+		EnvFrom:         []corev1.EnvFromSource{},
+		Command:         []string{"ping", "-c", "1", corePod.Status.PodIP},
+		Args:            nil,
+		SecurityContext: &corev1.SecurityContext{
+			AllowPrivilegeEscalation: &falseP,
+			RunAsNonRoot:             &trueP,
+			RunAsUser:                &userID,
+			SeccompProfile: &corev1.SeccompProfile{
+				Type: corev1.SeccompProfileTypeRuntimeDefault,
+			},
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{
+					corev1.Capability("ALL"),
+				},
+			},
+		},
+	}
+	podTemplate := corev1.PodTemplateSpec{
```

Comment:
we do have this action where we create a pod as well as a deployment: https://github.com/rancher/rancher/blob/d2ffe780f6bb73ccb10f373c92b8fda65b506f60/tests/v2/actions/workloads/deployment/deployment.go#L35 

We can update the container adding security context in there. so it can be for any future updates required to security context as well

Reply:
I think its overkill to have an action for this

Reply:
sorry didnt see this. I meant we do have an action for it already and can edit the container's spec to add security context for future reference.